src/hostnet/slirp.ml

```diff
@@ -239,14 +249,16 @@ struct
       tcp4:                     Stack_tcp.t;
       clock:                    Clock.t;
       mutable pending:          Tcp.Id.Set.t;
-      mutable last_active_time: float;
+      mutable last_active_time_ns: int64;
       (* Used to shutdown connections when the endpoint is removed from the switch. *)
       mutable established:      Tcp.Id.Set.t;
     }
     (** A generic TCP/IP endpoint *)
 
     let touch t =
-      t.last_active_time <- Unix.gettimeofday ()
+      t.last_active_time_ns <- Clock.elapsed_ns t.clock
```

Comment:
`Clock.elapsed_ns` calls into https://github.com/mirage/mirage-clock/blob/987d197802f80778b7b3daa642caee0c27a1ce43/unix/mclock.ml#L19 which is https://github.com/mirage/mirage-clock/blob/987d197802f80778b7b3daa642caee0c27a1ce43/unix/clock_stubs.c#L67 

According to https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time

> Returns current value of a clock that increments monotonically in tick units (starting at an arbitrary point), this clock does not increment while the system is asleep.